src/Resolve/graphtype.jl

```diff
@@ -1439,7 +1439,13 @@ function prune_graph!(graph::Graph)
         return pvers0[vmsk0[1:(end-1)]]
     end
     new_pvers = [compute_pvers(new_p0) for new_p0 = 1:new_np]
-    new_vdict = [Dict(vn => v0 for (v0,vn) in enumerate(new_pvers[new_p0])) for new_p0 = 1:new_np]
+    new_vdict = Vector{Dict{VersionNumber, Int}}(undef, length(new_pvers))
+    for new_p0 in eachindex(new_vdict)
+        new_vdict[new_p0] = Dict(vn => v0 for (v0,vn) in enumerate(new_pvers[new_p0]))
+    end
+    # The code above is essentially equivalent to
+    # new_vdict = [Dict(vn => v0 for (v0,vn) in enumerate(new_pvers[new_p0])) for new_p0 = 1:new_np]
```

Comment:
I think this whole chunk can used be written as

```
# explicitly writing out the loop since the generator equivalent caused type inference failure
```

Reply:
I will be AFK in the next few days. Could you please make the changes you suggested? Otherwise, I will revisit this some time in October.